### PR TITLE
`Blocks`, renaming and `Cerealize` refactoring.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi @mzhurovich ,

This looks big enough, so I'll hold my horses a bit with sending out **everything**, as you asked. :-)

[[ You may want to merge in https://github.com/c5t/Current/pull/216 first ]].

Changes:

* `Yoda` is now at the top level.
* `Blocks` is now at the top level, with `HTTP` and `URL`.
* `Bricks/cerealize` is now friendly with both simple and polymorphic types.

Notes:

* No 2nd level namespaces in `Blocks`, just `namespace blocks { ... }`. Looks clean to me.
* File lengths in `cerealize`'s golden tests are shortened JSON keys (I replaced `"value0"` by `"e"` and `"p"` for simple and polymorphic types respectively.
* I'd appreciate it if you look at header guards too.
* `CTFO_BE` works as expected, sending out in a separate request.

Thanks!
Dima

### CODE REVIEW

File: Blocks/HTTP/impl/posix_server.h [hunk 1]
```diff
@@ -25,8 +25,8 @@ SOFTWARE.
 // TODO(dkorolev): Refactor the code to disallow BODY-less POST requests.
 // TODO(dkorolev): Support receiving body via POST requests. Add a test for it.
 
-#ifndef BRICKS_NET_API_POSIX_SERVER_H
-#define BRICKS_NET_API_POSIX_SERVER_H
+#ifndef BLOCKS_HTTP_POSIX_SERVER_H
```
Comment (mzhurovich) [inline Blocks/HTTP/impl/posix_server.h]: `IMPL` missing here.

File: Blocks/HTTP/impl/posix_client.h [hunk 2]
```diff
@@ -25,26 +25,25 @@ SOFTWARE.
 // TODO(dkorolev): Handle empty POST body. Add a test for it.
 // TODO(dkorolev): Support receiving body via POST requests. Add a test for it.
 
-#ifndef BRICKS_NET_API_POSIX_CLIENT_H
-#define BRICKS_NET_API_POSIX_CLIENT_H
+#ifndef BLOCKS_HTTP_POSIX_CLIENT_H
+#define BLOCKS_HTTP_POSIX_CLIENT_H
```
Comment (mzhurovich) [inline Blocks/HTTP/impl/posix_client.h]: `IMPL` missing here.

File: Blocks/HTTP/impl/posix_server.h (hunk 1 above)
Comment (dkorolev) [inline Blocks/HTTP/impl/posix_server.h]: :+1: 896e4ac

File: Blocks/HTTP/impl/posix_client.h (hunk 2 above)
Comment (dkorolev) [inline Blocks/HTTP/impl/posix_client.h]: :+1: 896e4ac

**Pull request merged**